src/main/java/org/spongepowered/api/CatalogType.java

```diff
@@ -60,5 +64,13 @@
      * @return The human-readable name of this catalog type
      */
     String getName();
+    
+    @Override
+    default Text toText() {
```

Comment:
I just had a good example for this.

Lets assume you have a random `CatalogType` instance. You want to display it with green color.

How would you do that?
````java
return Text.of(TextColors.GREEN, instance.getName());
````

Looks easy, but there might be a `Tranlsation` for your type as well.
Okay, so we change it to this

````java
if (instance instanceOf Translatable) {
    return Text.of(TextColors.GREEN, instance.getTranslation());
} else {
    return Text.of(TextColors.GREEN, instance.getName());
}
````
There might be plugin provided types that don't have a translation although they are `Translatable`.

````java
if (instance instanceOf Translatable) {
    if (instance.getTranslation()==null) {
        return Text.of(TextColors.GREEN, instance.getName());
    } else {
        return Text.of(TextColors.GREEN, instance.getTranslation());
    }
} else {
    return Text.of(TextColors.GREEN, instance.getName());
}
````

Mhh, not very beautiful. 

"Oh I forgot if this instance if from Plugin XY then there is a pretty print method."

````java
if (instance instanceOf PluginABCType) {
   return Text.of(TextColors.GREEN, instance.getText());
} else if (instance instanceOf PluginXYType) {
   return Text.of(TextColors.GREEN, instance.getPrettyText());
} else if (instance instanceOf Translatable) {
    if (instance.getTranslation()==null) {
        return Text.of(TextColors.GREEN, instance.getName());
    } else {
        return Text.of(TextColors.GREEN, instance.getTranslation());
    }
} else {
    return Text.of(TextColors.GREEN, instance.getName());
}
````
Ugly...

and with my PR
````java
return Text.of(TextColors.GREEN, instance);
````

"But plugin XY does not implement that method (like I want)."

Do you know the plugin / Do you have some kind of class dependency?

"No"

Then you couldn't have formatted it properly by yourself as well.